sources/Core/Email/EmailLaminas.php

```diff
@@ -431,22 +417,13 @@ public function AddPart($sText, $sMimeType = Mime::TYPE_HTML)
 		$oNewPart = new Part($sText);
 		$oNewPart->encoding = Mime::ENCODING_8BIT;
 		$oNewPart->type = $sMimeType;
-		$this->m_oMessage->getBody()->addPart($oNewPart);
+
+		// setBody called only to refresh Content-Type to multipart/mixed
+		$this->m_oMessage->setBody($this->m_oMessage->getBody()->addPart($oNewPart));
```

Comment:
Similar as previous suggestion:
```suggestion
		$oBody->$this->m_oMessage->getBody()
		$oBody->addPart($oNewPart)
		$this->m_oMessage->setBody($oBody);
```

Reply:
- First added line should be `$oBody = $this->m_oMessage->getBody();`.
- Second added line is missing a `;`.
- First two lines can be combined to `$oBody = $this->m_oMessage->getBody()->addPart($oNewPart);`. The method returns the object. They call it "[fluent interface](https://en.wikipedia.org/wiki/Fluent_interface#PHP)". It is up to you, if you want 1, 2, or 3 lines. I am fine with either.
- I would keep the comment about why `setBody` is called at all. It is not obvious, because the previous `addPart` already modifies the messages body. What is being done is essentially `$this->m_oMessage->setBody($this->m_oMessage->getBody());`. The call is only necessary if the body was single-part previously. This ensures that the main `Content-Type` header gets updated accordingly ([here](https://github.com/laminas/laminas-mail/blob/2.26.x/src/Message.php#L410)). IMHO, this is a workaround in the library's deficiency.

Everything said applies to the first suggestion as well.

Reply:
> I would keep the comment about why `setBody` is called at all. It is not obvious, because the previous `addPart` already modifies the messages body.

That's exactly why I suggested to break that apart to make it more clear 😉
I was indeed a bit too quick, here's a better suggestion:
```suggestion
		$oBody = $this->m_oMessage->getBody();
		$oBody->addPart($oNewPart);
		$this->m_oMessage->setBody($oBody);
```
Could also be
```suggestion
		$oBody = $this->m_oMessage->getBody()->addPart($oNewPart);
		$this->m_oMessage->setBody($oBody);
```

I thought the idea was clear enough without being lexically/syntaxically correct.

Reply:
Which one do you prefer?

Reply:
It's up to @Combodo team to decide, I was just suggesting. I'm a person from the community just like you.

PS: You can join us on [Slack](https://join.slack.com/t/itopbycombodo/shared_invite/zt-uoat5wuf-C0g_1tqaeI9~EaqHxV7Q1Q) if you like.

Reply:
I have no preferences on this matter, I'll let others answer :sweat_smile: